components/support/ktx/src/test/java/mozilla/components/support/ktx/android/content/ContextTest.kt

```diff
@@ -44,4 +65,49 @@ class ContextTest {
 
         assertEquals(extensionFunctionResult, normalMethodResult)
     }
+
+    @Test
+    fun `isScreenReaderEnabled() returns true when TouchExploration enabled`() {
+        shadowOf(accessibilityManager).isTouchExplorationEnabled = true
+        context!!.isScreenReaderEnabled()?.let { assertTrue(it) }
+    }
+
+    @Test
+    fun `isScreenReaderEnabled() returns false when TouchExploration disabled`() {
+        shadowOf(accessibilityManager).isTouchExplorationEnabled = false
+        context!!.isScreenReaderEnabled()?.let { assertFalse(it) }
+    }
+
+    @Test
+    fun `getAccessibilityManager() provides a working instance`() {
+        shadowOf(accessibilityManager).isEnabled = true
+        assertTrue(accessibilityManager!!.isEnabled)
+        assertTrue(getAccessibilityManagerInstance().isEnabled)
+
+        shadowOf(accessibilityManager).isEnabled = false
+        assertFalse(accessibilityManager!!.isEnabled)
+        assertFalse(getAccessibilityManagerInstance().isEnabled)
+    }
+
+    @Test
+    fun `screenReaderSupport when AccessibiltyManager is not available`() {
+        // Assume AccessibilityManager is null
+        val spyContext = spy(context)
+        `when`(spyContext?.getAccessibilityManager()).thenReturn(null)
+        assertNull(spyContext?.isScreenReaderEnabled())
+
+        // Assume AccessibilityManager is NOT null
+        val accessibilityManager: AccessibilityManager = mock(AccessibilityManager::class.java)
+        `when`(spyContext?.getAccessibilityManager()).thenReturn(accessibilityManager)
+        `when`(accessibilityManager.isTouchExplorationEnabled).thenReturn(true)
+        val res = spyContext?.isScreenReaderEnabled()
+        assertNotNull(res)
+        assertTrue(res ?: false)
+    }
+
+    @Throws(Exception::class)
+    private fun getAccessibilityManagerInstance(): AccessibilityManager {
+        return ReflectionHelpers.callStaticMethod(AccessibilityManager::class.java, "getInstance",
+                ReflectionHelpers.ClassParameter.from(Context::class.java, RuntimeEnvironment.application))
+    }
```

Comment:
Why is reflection needed here?

Reply:
I used the roboelctric sample as a reference https://github.com/robolectric/robolectric/blob/master/robolectric/src/test/java/org/robolectric/shadows/ShadowAccessibilityManagerTest.java